pkg/scaling/resolver/scale_resolvers_test.go

```diff
@@ -541,3 +541,54 @@ func TestResolveDependentEnv(t *testing.T) {
 		})
 	}
 }
+
+func TestResolveEnvWithRestrictedSecretAccess(t *testing.T) {
+	tests := []struct {
+		name      string
+		expected  map[string]string
+		container *corev1.Container
+	}{
+		{
+			name: "env reference secret key",
+			container: &corev1.Container{
+				Env: []corev1.EnvVar{{
+					Name: envKey,
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: secretName,
+							},
+							Key: secretKey,
+						},
+					},
+				}},
+			},
+			expected: map[string]string{},
+		},
+		{
+			name: "env reference secret name",
+			container: &corev1.Container{
+				EnvFrom: []corev1.EnvFromSource{{
+					SecretRef: &corev1.SecretEnvSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: secretName,
+						},
+					},
+				}},
+			},
+			expected: map[string]string{},
+		},
```

Comment:
Is it possible to test "restricted secret access and namespace is keda" case?

Reply:
Hi @tobotg . When namespace is keda, we are trying to fetch secrets from secretsLIster. Any way we can mock secretLIster? I tried in local and I ended up getting below error after setting required env variables

panic: runtime error: invalid memory address or nil pointer dereference [recovered]

Any suggestions on this?

Reply:
Here you can see how we are mocking other stuff https://github.com/kedacore/keda/blob/bc921248e83d8c169d9ed9883e6c666979a9bcb2/Makefile#L174 - this is particular mocks scale interface of `k8s.io/clinet-go`. 

So the approach can be identical.

Reply:
Thanks @zroubalik for your suggestion. Have added test case for the same